trino/client.py

```diff
@@ -839,3 +769,110 @@ def decorated(*args, **kwargs):
         return decorated
 
     return wrapper
+
+
+class RowMapperFactory:
+    def __init__(self, columns):
+        if columns is None:
+            self.mapping_funcs = []
+        else:
+            self.mapping_funcs = [self._col_func(column['typeSignature']) for column in columns]
+
+    def map_rows(self, rows):
+        return [self._map_row(row) for row in rows]
+
+    def _map_row(self, row):
+        return [self._map_value(value, self.mapping_funcs[idx]) for idx, value in enumerate(row)]
+
+    def _map_value(self, value, col_mapping_func):
+        if value is None:
+            return None
+
+        try:
+            return col_mapping_func(value)
+        except ValueError as e:
+            error_str = f"Could not convert '{value}' into the associated python type"
+            raise trino.exceptions.TrinoDataError(error_str) from e
```

Comment:
Move to separate class `RowMapper`. If no col_mapping_funcs are provided in constructor, return the row value as is.

Reply:
Done